Jucer2Reprojucer/main.cpp

```diff
@@ -270,6 +270,51 @@ void writeUserNotes(LineWriter& wLn, const juce::ValueTree& valueTree)
   }
 }
 
+    juce::ValueTree sanitizeConfigurationNames(const juce::String& exporterName, const juce::ValueTree& exporterSettingsWithInvalidConfigNames)
+    {
+        // Removes invalid characters from configuration names and adds digits to the names until all
+        // configuration names within an exporter are unique.
+        // Valid config names match the regex ^[A-Za-z0-9_]*$
+
+        const char* validCharacters = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789_";
+
+        auto exporter = exporterSettingsWithInvalidConfigNames.createCopy();
+        juce::StringArray sanitizedConfigNames;
+
+        for (int i = 0; i < exporter.getNumChildren(); i++)
+        {
+            // remove invalid characters
+            auto configuration = exporter.getChild(i);
+            auto originalName = configuration.getProperty("name").toString();
+            auto sanitizedName = originalName.retainCharacters(validCharacters);
+
+            if (sanitizedName.isEmpty())
+            {
+                sanitizedName = "InvalidEmptyName";
+                std::cerr << "warning: Configuration '" << originalName << "' in exporter '" << exporterName
+                          << "' contained only invalid characters. The name was changed to '" << sanitizedName << "'" << std::endl;
+            }
+            else if (originalName != sanitizedName)
+                std::cerr << "warning: Configuration '" << originalName << "' in exporter '" << exporterName
+                          << "' contains invalid characters. The name was changed to '" << sanitizedName << "'" << std::endl;
+
+            // add digits until unique
+            int number = 1;
+            auto sanitizedNameWithOptionalNumber = sanitizedName;
+            while (sanitizedConfigNames.contains(sanitizedNameWithOptionalNumber))
+                sanitizedNameWithOptionalNumber = sanitizedName + juce::String(number++);
```

Comment:
This is a great idea! However, there is an edge case that this implementation doesn't cover: when exporters don't have the configurations in the same order, or when some configurations only exist on certain exporters.

For instance:
```yaml
- XCODE_MAC:
  - configurations:
    - name: "Config (-feature)"
      defines: "FEATURE=0"
    - name: "Config (+feature)"
      defines: "FEATURE=1"

- LINUX_MAKE:
  - configurations:
    - name: "Config (+feature)"
      defines: "FEATURE=1"
    - name: "Config (-feature)"
      defines: "FEATURE=0"
```
would be converted to 
```yaml
- XCODE_MAC:
  - configurations:
    - name: "Configfeature"
      defines: "FEATURE=0"
    - name: "Configfeature1"
      defines: "FEATURE=1"

- LINUX_MAKE:
  - configurations:
    - name: "Configfeature"
      defines: "FEATURE=1"
    - name: "Configfeature1"
      defines: "FEATURE=0"
```
which would break the matching of the configuration names between the exporters.

To solve this issue, we should collect all configuration names, make a mapping of original to sanitized configuration names, and then use that mapping when writing the configuration names to the `CMakeLists.txt` file.

Reply:
Hmm, I see. Yes, the matching would break. You're right, that should be avoided. I'll make the required changes.